packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -73,15 +79,25 @@ abstract class AssetBundle {
       throw FlutterError('Unable to load asset: $key'); // ignore: dead_code
     // 50 KB of data should take 2-3 ms to parse on a Moto G4, and about 400 μs
     // on a Pixel 4.
-    if (data.lengthInBytes < 50 * 1024) {
-      return utf8.decode(data.buffer.asUint8List());
+    if (data.lengthInBytes < 50 * 1024 && !unzip) {
+      return _utf8Decode(data);
     }
     // For strings larger than 50 KB, run the computation in an isolate to
     // avoid causing main thread jank.
-    return compute(_utf8decode, data, debugLabel: 'UTF8 decode for "$key"');
+    return compute(
+      unzip ? _utf8ZipDecode : _utf8Decode,
+      data,
+      debugLabel: 'UTF8 decode for "$key"',
+    );
+  }
+
+  static String _utf8ZipDecode(ByteData data) {
+    List<int> bytes = data.buffer.asUint8List();
+    bytes = gzip.decode(bytes);
```

Comment:
@yjbanov web question while Jonah is out. I can't do this on web right?

Reply:
`dart:io` is not available. However, the server and browser are expected to compress all data exchanged over the network, so we can just skip the decoding here if we're on the web.